applications/substrate-evm-adapter.md

```diff
@@ -0,0 +1,189 @@
+- `Hyperfridge` - https://github.com/element36-io/ocw-ebics
+- `ISO-8583 POC` - https://github.com/subclone/iso8583-chain
+- `polkadot-sdk` - https://github.com/paritytech/polkadot-sdk/pulls/dastansam
+- `Substrate StackExchange` - https://substrate.stackexchange.com/users/2005/dastansam
+
+### Team Code Repos
+
+- https://github.com/subclone/subeth
+
+Team members:
+
+- https://github.com/dastansam
+- https://github.com/dastanbeksamatov
+
+### Team LinkedIn Profiles (if available)
+
+- https://linkedin.com/in/dastanbek-samatov
+
+## Development Roadmap :nut_and_bolt:
+
+### Overview
+
+- **Total Estimated Duration:** 3.5 months
+- **Full-Time Equivalent (FTE):**  1 FTE
+- **Total Costs:** 30,000 USDT
+
+### Milestone 1 — ETH RPC Adapter
+
+- **Estimated duration:** 6 weeks
+- **FTE:**  1
+- **Costs:** 12,000 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| **0a.** | License | MIT |
+| **0b.** | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a user can (for example) spin up the adapter and connect it to the familiar developer tools. |
+| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| **0d.** | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| 1. | ETH-RPC Adapter | We will create a generic ETH-RPC Adapter service for Substrate chains. It will support the most essential ETH RPC calls to make it work with the existing tools. Main goals here will be ability to connect to Metamask, `web3.js`, read Substrate chain's pallets' state and support subscriptions. Adapter will have option of running local `smoldot` instance or connecting to remote RPC node. |
+| 2. | Deno module | We will create a Deno module that can connect to Substrate chain as an ETH-RPC adapter + light client. |
+| 3. | Javascript package | We will provide a javascript package that can connect to Substrate chain as an ETH-RPC adapter + light client. |
+| 4. | Unit tests | We will provide comprehensive unit tests for the adapter. |
+
+### Milestone 2 — EVM Adapter Pallet
+
+- **Estimated duration:** 9 weeks
+- **FTE:**  1
+- **Costs:** 18,000 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| **0a.** | License | MIT |
+| **0b.** | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a user can (for example) spin up one of our Substrate nodes and send test transactions, which will show how the new functionality works. |
+| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
```

Comment:
As you are developing a pallet which should be able to interact with other pallets I think you need to provide integration tests in addition to the unit tests.